devnet-sdk/devstack/telemetry/slog.go

```diff
@@ -0,0 +1,78 @@
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func WrapHandler(h slog.Handler) slog.Handler {
+	return &tracingHandler{
+		Handler: h,
+	}
+}
+
+type tracingHandler struct {
+	slog.Handler
+}
+
+func (h *tracingHandler) Handle(ctx context.Context, record slog.Record) error {
+	// Send log entries as events to the tracer
+	if h.Handler.Enabled(ctx, record.Level) {
+		span := trace.SpanFromContext(ctx)
+		if span.IsRecording() {
+			attrRecorder := &attrAccumulator{}
+			record.Attrs(func(a slog.Attr) bool {
+				attrRecorder.register(a)
+				return true
+			})
+			span.AddEvent(record.Message, trace.WithAttributes(attrRecorder.kv...))
+		}
+	}
+
+	// Conversely add tracing data to the local logs
+	spanCtx := trace.SpanContextFromContext(ctx)
+	if spanCtx.HasTraceID() {
+		record.AddAttrs(slog.String("trace_id", spanCtx.TraceID().String()))
+	}
+	if spanCtx.HasSpanID() {
+		record.AddAttrs(slog.String("span_id", spanCtx.SpanID().String()))
+	}
+	return h.Handler.Handle(ctx, record)
+}
+
+type attrAccumulator struct {
+	kv []attribute.KeyValue
+}
+
+func (ac *attrAccumulator) register(a slog.Attr) {
+	switch a.Value.Kind() {
+	case slog.KindAny:
+		ac.kv = append(ac.kv, attribute.String(a.Key, fmt.Sprintf("%v", a.Value.Any())))
+	case slog.KindBool:
+		ac.kv = append(ac.kv, attribute.Bool(a.Key, a.Value.Bool()))
+	case slog.KindDuration:
+		ac.kv = append(ac.kv, attribute.String(a.Key, a.Value.Duration().String()))
+	case slog.KindFloat64:
+		ac.kv = append(ac.kv, attribute.Float64(a.Key, a.Value.Float64()))
+	case slog.KindInt64:
+		ac.kv = append(ac.kv, attribute.Int64(a.Key, a.Value.Int64()))
+	case slog.KindString:
+		ac.kv = append(ac.kv, attribute.String(a.Key, a.Value.String()))
+	case slog.KindTime:
+		ac.kv = append(ac.kv, attribute.String(a.Key, a.Value.Time().String()))
+	case slog.KindUint64:
+		// we might be in trouble here, but opentelemetry doesn't support uint types
```

Comment:
Should we try detect when the issue occurs (perhaps by converting back to a uint and comparing with the original?) and then log a warning/error when this happens?

Reply:
good point, it was lazy of me :)